app/controllers/coronavirus_form/date_of_birth_controller.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class CoronavirusForm::DateOfBirthController < ApplicationController
+  include ActionView::Helpers::SanitizeHelper
+  include FieldValidationHelper
+
+  def show
+    session["date_of_birth"] ||= {}
+    render "coronavirus_form/#{PAGE}"
+  end
+
+  def submit
+    session["date_of_birth"] ||= {}
+    session["date_of_birth"]["day"] = sanitize(params.dig("date_of_birth", "day")).presence
+    session["date_of_birth"]["month"] = sanitize(params.dig("date_of_birth", "month")).presence
+    session["date_of_birth"]["year"] = sanitize(params.dig("date_of_birth", "year")).presence
+
+    invalid_fields = validate_date_fields(
+      session["date_of_birth"]["year"],
+      session["date_of_birth"]["month"],
+      session["date_of_birth"]["day"],
+      "date_of_birth",
+    )
+
+    if invalid_fields.any?
+      flash.now[:validation] = invalid_fields
+      render "coronavirus_form/#{PAGE}"
+    elsif session["check_answers_seen"]
+      redirect_to controller: "coronavirus_form/check_answers", action: "show"
+    else
+      redirect_to controller: "coronavirus_form/#{NEXT_PAGE}", action: "show"
+    end
+  end
+
+private
+
+  PAGE = "date_of_birth"
+  NEXT_PAGE = "start" # TODO changeme
+
+  def previous_path
+    "/" # TODO
```

Comment:
I think contact_details has been merged now, can be this, no?
```suggestion
    coronavirus_form_contact_details_path
```